### PR TITLE
Add support of guzzlehttp/guzzle:^7.0 to composer.json

### PR DESCRIPTION
**Pull request by drdaemos**
Hey @stevenmaguire 👋🏻 

We are using this library in our project, but its' locked guzzlehttp/guzzle version doesn't allow us to upgrade other dependencies to their maintainable versions. This PR changes composer.json to support guzzle v7 and also updates tests to work with it.

Would you kindly consider to include this into upstream?

### CODE REVIEW

File: tests/ClientTest.php [hunk 1]
```diff
@@ -125,9 +128,8 @@ protected function prepareFor($method, $path, $query = "", $payload = [], $statu
             $client->shouldReceive('send')->with($request, $requestOptions)->andReturn($response);
         } else {
             $badRequest = m::mock(RequestInterface::class);
-            if ($response) {
-                $response->shouldReceive('getReasonPhrase')->andReturn("");
-            }
+            $response->shouldReceive('getReasonPhrase')->andReturn("");
+
             $exception = new BadResponseException('test exception', $badRequest, $response);
```
Comment (drdaemos) [inline tests/ClientTest.php:136]: $response now is required to be a `ResponseInterface` instance

**Comment by stevenmaguire** [comment]
Thanks for this @drdaemos! I appreciate the contribution, especially the updates to the tests. 

Unfortunately, you might notice that the CI config is a bit...out of date. I want to get CI and code checks back up and running on the repo then I will merge this through and bump the major version number.

**Comment by stevenmaguire** [comment]
I've pulled in some tooling to take advantage of Github actions. That tooling attempts to fix code style issues. Those fixes have likely pulled your branch out of auto-resolve-conflicts state. 

Would you mind updating your branch to resolve these conflicts and we'll proceed with the merge and the CI updating?

**Comment by drdaemos** [comment]
@stevenmaguire thanks for responding so fast to this!

I've just pushed an update to my branch that resolves the conflict.

**Comment by kibertoad** [comment]
@stevenmaguire main is failing:
```
1) Stevenmaguire\Services\Trello\Tests\ClientTest::testAddCardAttachment
Error: Class 'GuzzleHttp\Psr7\Utils' not found

/home/runner/work/trello-php/trello-php/src/Http.php:76
/home/runner/work/trello-php/trello-php/src/Http.php:135
/home/runner/work/trello-php/trello-php/src/Http.php:214
/home/runner/work/trello-php/trello-php/src/Traits/ApiMethodsTrait.php:388
/home/runner/work/trello-php/trello-php/tests/ApiTestTrait.php:1235
phpvfscomposer:///home/runner/work/trello-php/trello-php/vendor/phpunit/phpunit/phpunit:60

2) Stevenmaguire\Services\Trello\Tests\ClientTest::testUpdateCardCustomField
Error: Class 'GuzzleHttp\Psr7\Utils' not found

/home/runner/work/trello-php/trello-php/src/Http.php:246
/home/runner/work/trello-php/trello-php/src/Traits/ApiMethodsTrait.php:388
/home/runner/work/trello-php/trello-php/tests/ApiTestTrait.php:1344
phpvfscomposer:///home/runner/work/trello-php/trello-php/vendor/phpunit/phpunit/phpunit:60
```

**Comment by stevenmaguire** [comment]
@kibertoad that is not helpful.

> Unfortunately, you might notice that the CI config is a bit...out of date. I want to get CI and code checks back up and running on the repo then I will merge this through and bump the major version number.

> Would you mind updating your branch to **resolve these conflicts and we'll proceed with the merge and the CI updating?**

Is it not obvious to you that this repo is undergoing some updates?

**Comment by kibertoad** [comment]
Sorry.

**Pull request merged**